### DISCUSSION TITLE
Import source code of a Component

### DISCUSSION THREAD

**Discussion by laruiss** [vuepress/core#131] [Q&A] [answered]
Hi, I would like to port some features of this plugin for VuePress1 : [vuepress-plugin-vue-example](https://vuepress-plugin-vue-example.netlify.app), which uses `clientDynamicModules` (to load source code of a component).

In the docs, I read:
`clientDynamicModules`: removed, use `app.writeTemp()` in `onPrepared` hook

Where can I read more about that `app.writeTemp()`?

**Comment by meteorlxy** [comment, marked as answer]
It would be listed in the Node API documentation, which is not finished yet.

In fact, the `clientDynamicModules` is just writing a temp file, i.e.:

```js
clientDynamicModules: () => ({
  name: 'foo.js',
  content: 'export default {}',
})
```

equals to:

```js
async onPrepared(app) {
  await app.writeTemp('foo.js', 'export default {}')
}
```

You can use it in client via `@temp` alias:

```js
import foo from '@temp/foo'
```

They are literally equal, so that's the reason why we remove the `clientDynamicModules` hook in v2 - it's redundant.

> The `app.writeTemp` (or called `ctx.writeTemp`) also exists in v1
